backend/app/db/dataset/version.py

```diff
@@ -165,3 +173,298 @@ async def _freeze_dataset_visualization(
+
+    if (file := await FileDB.get(PydanticObjectId(file_id))) is not None:
+        # delete from elasticsearch
+        delete_document_by_id(es, settings.elasticsearch_index, str(file_id))
+
+        # TODO: Deleting individual versions will require updating version_id in mongo, or deleting entire document
+        await FileVersionDB.find(FileVersionDB.file_id == ObjectId(file_id)).delete()
+
+        # delete metadata
+        await MetadataDB.find(
+            MetadataDB.resource.resource_id == ObjectId(file_id),
+            MetadataDB.resource.collection == "files",
+        ).delete()
+
+        # delete thumbnail
+        await _delete_thumbnail(file, fs)
+
+        # delete visualization
+        await _delete_visualizations(file, fs)
+
+        # delete file raw bytes if not used anywhere else
+        await _delete_file(file, fs)
+
+    else:
+        raise HTTPException(status_code=404, detail=f"File {file_id} not found")
+
+
+async def remove_frozen_file_entry(frozne_file_id: Union[str, ObjectId], fs: Minio):
+    """Remove Frozen FileDB object into MongoDB, Minio, and associated metadata and version information."""
+    if (
+        frozen_file := await FileFreezeDB.get(PydanticObjectId(frozne_file_id))
+    ) is not None:
+        # delete metadata
+        await MetadataFreezeDB.find(
+            MetadataFreezeDB.resource.resource_id == ObjectId(frozne_file_id),
+            MetadataFreezeDB.resource.collection == "files",
+        ).delete()
+
+        # delete thumbnail
+        await _delete_frozen_thumbnail(frozen_file, fs)
+
+        # delete visualization
+        await _delete_frozen_visualizations(frozen_file, fs)
+
+        # if all above succeeded, delete from mongo
+        await _delete_frozen_file(frozen_file, fs)
+    else:
+        raise HTTPException(
+            status_code=404, detail=f"Released File {frozne_file_id} not found"
+        )
+
+
+async def remove_local_file_entry(file_id: Union[str, ObjectId], es: Elasticsearch):
+    """Remove FileDB object into MongoDB, Minio, and associated metadata and version information."""
+    if (file := await FileDB.get(PydanticObjectId(file_id))) is not None:
+        # delete from elasticsearch
+        delete_document_by_id(es, settings.elasticsearch_index, str(file_id))
+
+        # delete metadata
+        await MetadataDB.find(
+            MetadataDB.resource.resource_id == PydanticObjectId(file_id)
+        ).delete()
+
+        # delete thumbnail
+        await _delete_thumbnail(file, None)
+
+        # delete visualization
+        await _delete_visualizations(file, None)
+
+        # if all above succeeded
+        await _delete_file(file, None)
+
+    else:
+        raise HTTPException(status_code=404, detail=f"File {file_id} not found")
+
+
+async def remove_frozen_local_file_entry(frozen_file_id: Union[str, ObjectId]):
+    """Remove FileDB object into MongoDB, Minio, and associated metadata and version information."""
```

Comment:
Are you deleting the file from the Minio here in this function?

Reply:
All these functions look very similar. Can you add a little more doc?
I am not able to understand the requirement for 4 separate functions apart from the fact that 2 of them are for frozen files.

Reply:
Yes I will add more descriptions. 
This is for local files which does not have minio backup. I didn't implement that feature so I am not 100% sure what's the usecase. 
But to keep it consistent with the original implementation,  I'm not passing in the `fs` hence not deleting any raw bytes.
```
await _delete_frozen_file(frozen_file, None)
```

Reply:
doc strings added.